third_party/terraform/utils/config.go.erb

```diff
@@ -158,20 +146,31 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	// See ClientWithAdditionalRetries
 	retryTransport := NewTransportWithDefaultRetries(loggingTransport)
 
+	// 4. Header Transport - outter wrapper to inject additional headers we want to apply
+	// before making requests
+	headerTransport := headerTrasportLayer(retryTransport)
+
 	// Set final transport value.
 	client.Transport = retryTransport
 
 	// This timeout is a timeout per HTTP request, not per logical operation.
 	client.Timeout = c.synchronousTimeout()
 
+	if len(c.Scopes) == 0 {
```

Comment:
Why move this down? Doesn't the oauth transport need the tokensource as a parameter?

Reply:
yes you're right, sorry was reorganizing the setting of c.~ .  testing a revising